### PR TITLE
Streamlined decay creation.

### PR DESCRIPTION
**Pull request by apuignav**
Basically, now particles get a `mass`, which is either a number or a function. The function gets min, max and n_events as input, and it allows you to generate any resonance with the proper limits.

TODO: test if it works properly with two resonances, but should be fine.

### CODE REVIEW

File: tfphasespace/tfphasespace.py [hunk 1]
```diff
@@ -25,117 +25,201 @@
...
+
+    A `Particle` must have a `name`, which is ensured not to clash with any others in
+    the decay chain.
+    It may also have:
+        + Mass, which can be either a number or a function to generate it according to
+        a certain distribution. In this case, the particle is not considered as having a
+        fixed mass and the `has_fixed_mass` method will return False.
+        + Children, ie, decay products, which are also `Particle` instances.
+
+    """
```
Comment (mayou36) [inline tfphasespace/tfphasespace.py]: Probably move that to `__init__` so that we can document the parameters as well?

Comment (mayou36) [reply tfphasespace/tfphasespace.py]: reconsidering it's good as it is, sorry

**Comment by apuignav** [comment]
Thanks! This should be ready to go?

**Comment by mayou36** [comment]
Jep. What about the TODO?

**Comment by apuignav** [comment]
I think we leave it for v2

**Comment by mayou36** [comment]
Alright, good from my side. Any changes left? Else merge

**Comment by apuignav** [comment]
Nah, let's merge.

**Pull request merged**